docs/topics/testing/tools.txt

```diff
@@ -570,6 +570,67 @@ can access these properties as part of a test condition.
             session['somekey'] = 'test'
             session.save()
 
+Setting the language
+--------------------
+
+When testing web applications which support internationalization and
+localizations, the language for a test client request can be set within a test
+case. The method of doing so depends on whether the
+:class:`~django.middleware.locale.LocaleMiddleware` is activated or not.
+
+If the middleware has been activated then the language can be set by
+specifying the language preference. This can be done by either creating a
+cookie with a name of :setting:`LANGUAGE_COOKIE_NAME` and a value of the
+language code, or by including the ``Accept-Language`` HTTP header in the
+request. More details on the topic of language preference can be found in the
+:ref:`internationalization documentation <how-django-discovers-language-preference>`
+and an example of both approaches is given below::
+
+    def test_language_using_cookie(self):
+        expected_content = b"Bienvenue sur mon site."
+
+        user_language = 'fr'
+        # set the desired language via the language code cookie
+        self.client.cookies.load({settings.LANGUAGE_COOKIE_NAME: user_language)
+
+        response = self.client.get('/polls/')
+        actual_content = response.content
+
+        self.assertEqual(actual_content, expected_content)
+
+    def test_language_using_header(self):
+        expected_content = b"Bienvenue sur mon site."
+
+        user_language = 'fr'
+        # set the desired language via the Accept-Language HTTP header
+        response = self.client.get('/polls/',
+                                   None,
+                                   HTTP_ACCEPT_LANGUAGE=user_language)
```

Comment:
We usually favor the following indentation

```python
response = self.client.get(
    '/polls/',
    HTTP_ACCEPT_LANGUAGE=user_language,
)
```
Note that you don't need to pass `None` as it's already the default value for the `data` kwarg.